Use a more reasonable fallback directory for fonts on Windows.

The existing fallback points to a directory that no longer exists by default
which causes warnings like the following on run-time:

```
QFontDatabase: Cannot find font directory C:/msys64/mingw64/qt6/libs/fonts.
Note that Qt no longer ships fonts. Deploy some (from
https://dejavu-fonts.github.io/ for example) or switch to fontconfig.
```

diff -urN qtbase-everywhere-src-6.12.0/src/gui/text/qplatformfontdatabase.cpp.orig qtbase-everywhere-src-6.12.0/src/gui/text/qplatformfontdatabase.cpp
--- qtbase-everywhere-src-6.12.0/src/gui/text/qplatformfontdatabase.cpp.orig
+++ qtbase-everywhere-src-6.12.0/src/gui/text/qplatformfontdatabase.cpp
@@ -362,9 +362,17 @@
 {
     QString fontpath = qEnvironmentVariable("QT_QPA_FONTDIR");
     if (fontpath.isEmpty()) {
+#ifdef Q_OS_WIN
+        QString windir = qEnvironmentVariable("WINDIR");
+        if (windir.isEmpty())
+            fontpath = "C:/Windows/Fonts"_L1;
+        else
+            fontpath = windir + "/Fonts"_L1;
+#else
         QStringList paths = QLibraryInfo::paths(QLibraryInfo::LibrariesPath);
         if (!paths.isEmpty())
             fontpath = paths.takeFirst() + "/fonts"_L1;
+#endif
     }
 
     return fontpath;
